exercises/concept/dancing-dots/.meta/exemplar.ex

```diff
@@ -0,0 +1,47 @@
+defmodule DancingDots.Animation do
+  @type dot :: DancingDots.Dot.t()
+  @type opts :: keyword
+  @type error :: any
+  @type frame_number :: pos_integer
+
+  @callback init(opts :: opts) :: {:ok, opts} | {:error, error}
+  @callback handle_frame(dot :: dot, n :: frame_number, opts :: opts) :: dot
+
+  defmacro __using__(_) do
+    quote do
+      @behaviour DancingDots.Animation
+      def init(opts), do: {:ok, opts}
+      defoverridable init: 1
+    end
+  end
+end
+
+defmodule DancingDots.Flicker do
+  use DancingDots.Animation
+
+  @impl DancingDots.Animation
+  def handle_frame(dot, frame_number, _opts) do
+    opacity = if rem(frame_number, 4) == 0, do: dot.opacity / 2, else: dot.opacity
+    %{dot | opacity: opacity}
+  end
+end
+
+defmodule DancingDots.Zoom do
+  use DancingDots.Animation
+
+  @impl DancingDots.Animation
+  def init(opts) do
+    velocity = Keyword.get(opts, :velocity)
+
+    if is_number(velocity) do
+      {:ok, [velocity: velocity]}
+    else
+      {:error, "Expected required option :velocity to be a number, got: #{inspect(velocity)}"}
```

Comment:
I would suggest to add one type of errors here: "Missing option :velocity". And some tests for that.

Reply:
I'm not sure I understand the suggestion, are you saying there need to be two different errors, one for when velocity was not given, and one for when it's given but not valid? There already is a test for when it's missing:
```elixir
      assert DancingDots.Zoom.init([]) ==
               {:error, "Expected required option :velocity to be a number, got: nil"}
```

Reply:
Yes, I was suggesting two different errors, but actually a simple rephrasing might do.

If no `:velocity` option is given, the error message says "Expected required option `:velocity` to be a number", which feels like you passed something wrong, not nothing at all.

How about:
"The `:velocity` option is required, and its value must be a number. Got: nil"
It's a small change, but it feels a bit better maybe?

Reply:
Totally, that does sound better. Here: [`ccfcabe` (#1103)](https://github.com/exercism/elixir/pull/1103/commits/ccfcabeb06e86eda16a84d68ffea071b3b5951f4)